qutebrowser/mainwindow/mainwindow.py

```diff
@@ -490,18 +486,12 @@ def _connect_signals(self):
     @pyqtSlot(bool)
     def _on_fullscreen_requested(self, on):
         if on:
-            self._restore_fullscreen = self.isFullScreen()
-            self._restore_maximized = self.isMaximized()
+            self._state_before_fullscreen = self.windowState()
             self.showFullScreen()
-        elif not self._restore_fullscreen:
-            if self._restore_maximized:
-                self.showMaximized()
-            else:
-                self.showNormal()
```

Comment:
Replacing this with an unconditioned `setWindowState` has the consequence that if I go to fullscreen, enter fullscreen video, go out of fullscreen with `<F11>` (or `<mod+f>` in case of my WM) -- note that the video is still "fullscreen" inside the qutebrowser window -- and finally exit the video fullscreen, then qutebrowser goes into fullscreen again. That seems weird, so I originally called `showNormal` only when fullscreen was not going to be restored.